src/Server/HTTP/HTTPServerResponse.cpp

```diff
@@ -53,75 +49,36 @@ std::shared_ptr<WriteBuffer> HTTPServerResponse::send()
         // but if we do, then it is safer to close the connection at the end
         setKeepAlive(false);
 
-        // Send header
-        Poco::Net::HTTPHeaderOutputStream hs(session);
-        write(hs);
         stream = std::make_shared<AutoFinalizedWriteBuffer<WriteBufferFromPocoSocket>>(session.socket(), write_event);
     }
     else if (getChunkedTransferEncoding())
     {
-        // Send header
-        Poco::Net::HTTPHeaderOutputStream hs(session);
-        write(hs);
         stream = std::make_shared<AutoFinalizedWriteBuffer<HTTPWriteBufferChunked>>(session.socket(), write_event);
     }
     else if (hasContentLength())
     {
-        // Send header
-        Poco::Net::HTTPHeaderOutputStream hs(session);
-        write(hs);
         stream = std::make_shared<AutoFinalizedWriteBuffer<HTTPWriteBufferFixedLength>>(session.socket(), getContentLength(), write_event);
     }
     else
     {
         setKeepAlive(false);
-        // Send header
-        Poco::Net::HTTPHeaderOutputStream hs(session);
-        write(hs);
```

Comment:
it is `HTTPResponse::write` method. It is not virtual method.
It calls `HTTPResponse::beginWrite` method.

But `HTTPServerResponse` has its own `beginWrite` method. That method should be used here. 
It is ugly. May be it is better to make a refactoring here without backporting it to the releases branches.